cql3/expr/expression.cc

```diff
@@ -1123,14 +1123,18 @@ std::ostream& operator<<(std::ostream& os, const expression::printer& pr) {
                         to_printer(cma.column));
             },
             [&] (const function_call& fc)  {
-                std::visit(overloaded_functor{
-                    [&] (const functions::function_name& named) {
-                        fmt::print(os, "{}({})", named, fmt::join(fc.args | transformed(to_printer), ", "));
-                    },
-                    [&] (const shared_ptr<functions::function>& anon) {
-                        fmt::print(os, "<anonymous function>({})", fmt::join(fc.args | transformed(to_printer), ", "));
-                    },
-                }, fc.func);
+                if (is_token_function(fc)) {
+                    fmt::print(os, "token({})", fmt::join(fc.args | transformed(to_printer), ", "));
+                } else {
+                    std::visit(overloaded_functor{
+                        [&] (const functions::function_name& named) {
+                            fmt::print(os, "{}({})", named, fmt::join(fc.args | transformed(to_printer), ", "));
+                        },
+                        [&] (const shared_ptr<functions::function>& anon) {
+                            fmt::print(os, "<anonymous function>({})", fmt::join(fc.args | transformed(to_printer), ", "));
```

Comment:
I'm not a fan of this "anonymous function" (see also https://github.com/scylladb/scylladb/issues/10632#issuecomment-1374875396). After all, the user somehow called this function in CQL, so there must be some CQL string that can be remembered together with the call and be used when we want to print this call.

But I guess this is a pre-existing problem. At least you are fixing it for token() which is what bothered me in that issue. I don't know in what other cases it can appear.

Reply:
I'm not sure why the original code had the `<anonymous function>`. functions::function has a name method: 
```c++
virtual const function_name& name() const = 0;
```

Maybe some internal functions have empty names because they were never meant to be printed? Then the user would get confused because they see `(arg)` instead of `<anonymous function>(arg)`. This is something that would be good to fix, but probably not in this PR.